pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, fetchFromGitHub
+, callPackage
+, crystal
+, libadwaita
+, gtk3
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+let
+  gi_crystal = callPackage gi-crystal/default.nix { };
+in
+crystal.buildCrystalPackage rec {
+  pname = "Collision";
+  version = "3.5.0";
+
+  src = fetchFromGitHub {
+    owner = "GeopJr";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    name = pname;
+    hash = "sha256-YNMtiMSzDqBlJJTUntRtL6oXg+IuyAobQ4FYcwOdOas=";
+  };
+
+
+  # Crystal compiler has a strange issue with OpenSSL. The project will not compile due to
+  # main_module:(.text+0x6f0): undefined reference to `SSL_library_init'
+  # There is an explanation for this https://danilafe.com/blog/crystal_nix_revisited/
+  # Shortly, adding pkg-config to buildInputs along with openssl fixes the issue.
+
+  nativeBuildInputs = [ gtk3 wrapGAppsHook4 pkg-config ];
+  buildInputs = [ libadwaita openssl libxml2 ];
+
+  format = "make";
+
+  patches = [ ./make.patch ./shard.patch ];
+  shardsFile = ./collision-shards.nix;
+
+  doInstallCheck = false;
+
+  buildPhase = ''
+    mkdir lib/gi-crystal
+    cp -r ${gi_crystal}/* lib/gi-crystal
+
+    shards build -Dpreview_mt --release --no-debug
```

Comment:
This (along with the shard.patch) can be replaced with:

`crystal build ./src/collision.cr -Dpreview_mt --release --no-debug`

Reply:
Changed to 
```
mkdir bin
crystal build ./src/collision.cr -o bin/collision -Dpreview_mt --release --no-debug
```
because `crystal build` create output file in the same directory and `make install` can't find the binary.